client/state/posts/utils.js

```diff
@@ -21,22 +21,31 @@ export function getNormalizedPostsQuery( query ) {
 
 /**
  * Returns a serialized posts query, used as the key in the
- * `state.posts.siteQueries` state object.
+ * `state.posts.queries` state object.
  *
- * @param  {Object} query Posts query
- * @return {String}       Serialized posts query
+ * @param  {Object} query  Posts query
+ * @param  {Number} siteId Optional site ID
+ * @return {String}        Serialized posts query
  */
-export function getSerializedPostsQuery( query = {} ) {
-	return JSON.stringify( getNormalizedPostsQuery( query ) ).toLowerCase();
+export function getSerializedPostsQuery( query = {}, siteId ) {
```

Comment:
>Do we ever parse the query again? If not, we could consider generating a shorter key.

This is a one-way conversion, yes, so it's feasible that we shorten the key. Which technique would you suggest? If we had `md5` easily available, that might be a good choice, but I'd hate to add it to the bundle if we're not already using it elsewhere.